### PR TITLE
I deleted our idle sprites

### PR DESCRIPTION
**Pull request by Booplicate**
Implemented a new system for idle sprite management . Instead of ATL, this uses its own class `MASMoniIdleDisp`. 

Why?
- `ATL` is buggy. For example it likes to restart when we don't need, but refuses to restart when we do need it to. It also has issues with other transforms and transitions.
- `ATL` gives no control over it, the new disp allows to change exp with one method.
- With `ATL` we can't really get status of `ATL`, the disp allows us to get current exp, its duration, and much more.
- For `ATL` we kinda had to duplicate code, but now adding a new expression is a matter of writing one line.
- `ATL` would be a pain to use with the mood system. This disp is kinda a step towards moods.
- I think my disp is more optimized than `ATL` + `ConditionSwitch`.

The new disp completely mimics the old ATL regarding how it looks. Although, for ATL sprites "total weight" was always `1`, for the disp total weight depends on the weight of all exps. So this may need to be adjusted, since I used the old values for weights.

New:
- `MASMoniIdleDisp` - the new displayable to handle idle sprite.
- `MASMoniIdleExp` - a class to represent a single expression.
- `MASMoniIdleExpGroup` - a class to represent a group of exps. This is used when we need to show multiple exp one by one. It's a subclass of `MASMoniIdleExp`, so they share some props and methods (and basically can be treated the same).
- `MASMoniIdleExpRngGroup` - a subclass of `MASMoniIdleExpGroup` for rng based img selection.
- New return key: `idle_exp` can be used to force a temp idle exp. The exp can be generated on the fly or being retrieved by a tag. For example: `return "idle_exp: 5eubsa, 5"` or `return "idle_exp: happy"`. If the tag doesn't exist, nothing will happen. If there's multiple exp with the same tag, we use weighted choice to select one. I haven't added any predefined groups yet.

Changes:
- I added some more idle sprites. If we want some more, then it's good time to add them.
- `monika idle` is now predicted after the splash screen.
- We can use `monika idle` for spaceroom dissolving again. Hooray for no `ATL` seizures.

### Testing:
- I had to change the wink transform, I haven't noticed any issues, but I'd ask to pay some attention to it.
- Please, verify that we have good idle sprites. Especially their parameters like weights/conditionals/exp codes.
- Verify loading times are fine, no freezes during switching from topic into idle, etc.
- Verify that `conditional`'d exps are only being shown if the condition is met, same for `aff_range`.
- Give the new system a try. Force a new exp, try to add or remove one during runtime.
- Play with the return keys, make sure we have everything we want.
- Verify no stranger things happen. Ideally this disp should fix all the issues with had with idle sprites.

### CODE REVIEW

**Comment by Booplicate** [comment]
@ThePotatoGuy Idk why Travis doesn't pass, we may need some adjustments to it.

**Comment by ThePotatoGuy** [comment]
did you have a replacement for 6atl_leftandright? that is also used in crash greeting.

regardless, you should alias the atl sprites for at least one version to prevent crashes.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]
the build is failing because you defined `monika idle` image at init 500, but labels are parsed at init 0. the image needs to be defined earlier than 0.

also can you move the disp to sprite-chart in a new python early block?

**Comment by Booplicate** [comment]
RenPy defined images at init 500, but sure thing.

**Comment by ThePotatoGuy** [comment]
oh thats actually probably because the spritechecker doesnt consider an indented sprite normal. just define the image on the base indent.

**Comment by Booplicate** [comment]
I'm dumb, we can't define the image early, it needs aff functions. We *have* to define it later.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]

File: Monika After Story/game/sprite-chart.rpy [hunk 1]
```diff
@@ -6891,6 +6891,1036 @@ init -3 python:
...
+                exps = [exps]
+            self.exps = exps
+
+            if max_uses is None:
+                max_uses = len(self.exps)
+            self.max_uses = max_uses
+            self.uses = 0
+
+            self.__finish_init(aff_range, conditional, weight, tag, repeatable, add_to_tag_map)
+            self.__validate_children_aff_range()
```
[the remark this answers is not in the document]
Comment (Booplicate) [reply Monika After Story/game/sprite-chart.rpy:7345]: Eh, not sure why I even did this.

File: Monika After Story/game/event-handler.rpy [hunk 2]
```diff
@@ -1425,6 +1426,9 @@ init -1 python in evhand:
         "unlock_prompt",
     ]
 
+    # Used to catch "idle_exp: EXP, DURATION" or "idle_exp: TAG"
+    RET_KEY_PATTERN_IDLE_EXP = re.compile(r"(?:(?<=\|)|(?<=^))\s*idle_exp\s*:\s*(?:(?P<exp>\d[a-z]{3,13})\s*,\s*(?P<duration>\d)|(?P<tag>\w+))\s*(?:(?=\|)|(?=$))")
```
Comment (multimokia) [inline Monika After Story/game/event-handler.rpy:1430]: While it's unlikely we'll need to use something over 10 seconds. We should still account for it.
```suggestion
    RET_KEY_PATTERN_IDLE_EXP = re.compile(r"(?:(?<=\|)|(?<=^))\s*idle_exp\s*:\s*(?:(?P<exp>\d[a-z]{3,13})\s*,\s*(?P<duration>\d+)|(?P<tag>\w+))\s*(?:(?=\|)|(?=$))")
```

**Review by multimokia — CHANGES_REQUESTED** [review]

**Review by ThePotatoGuy — DISMISSED** [review]

**Review by multimokia — APPROVED** [review]

**Review by ThePotatoGuy — APPROVED** [review]

**Pull request merged**